nautobot/users/api/views.py

```diff
@@ -63,6 +69,19 @@ def authentication_classes(self):
         classes = super().authentication_classes
         return classes + [BasicAuthentication]
 
+    @action(methods=["POST"], detail=False, permission_classes=[AllowAny])
+    def authenticate(self, request):
+        serializer = serializers.UserLoginSerializer(data=request.data, context=self.get_serializer_context())
+        serializer.is_valid(raise_exception=True)
+        user = serializer.validated_data["user"]
+        login(request, user=user)
+        return Response(status=200)
+
+    @action(methods=["GET"], detail=False, permission_classes=[AllowAny])
+    def logout(self, request):
+        logout(request)
+        return Response(status=200)
+
```

Comment:
Is there a reason we can't just `POST` to the existing login views? We'll also need to handle the SSO route as well.

Reply:
That would need a csrf token to be placed in the form, just guessed that's not "API" way of doing things,
But illd try it out

Reply:
I just attempted it, and I ran into this issue: The status codes 302[OK] and 200[Failure] indicate whether a request was successful or not, but there is no method to manage 302 in "axios."

Reply:
Confirmed this locally... eww...

Reply:
Last feedback item: Is there a reason we need to set `permission_classes=[AllowAny]`?

Reply:
For logout, that is a mistake, for login yes

Reply:
Is this because it's nested under the tokenviewset?

Reply:
Yes

Reply:
OK, can you toss in a `TODO` to potentially move elsewhere? We either create a user login viewset or fix the weird error codes in the default path.